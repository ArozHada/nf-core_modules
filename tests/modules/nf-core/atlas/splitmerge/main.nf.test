nextflow_process {

    name "Test Process ATLAS_SPLITMERGE"
    script "modules/nf-core/atlas/splitmerge/main.nf"
    process "ATLAS_SPLITMERGE"
    config "tests/modules/nf-core/atlas/splitmerge/nextflow.config"
    tag "atlas_splitmerge"

    test("Human Illumina PE Sorted BAM") {

        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = [ [ id:'test', single_end:false ],
                             file(params.test_data['homo_sapiens']['illumina']['test_paired_end_sorted_bam'], checkIfExists: true),
                             file(params.test_data['homo_sapiens']['illumina']['test_paired_end_sorted_bam_bai'], checkIfExists: true),
                             file(params.test_data['homo_sapiens']['illumina']['read_group_settings_txt'], checkIfExists: true),
                             []
                           ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(path("$outputDir/atlas/versions.yml"),
                                  path("$outputDir/atlas/test.paired_end.sorted_ignoredReads.txt.gz")).match() },
                { assert new File("$outputDir/atlas/test.paired_end.sorted_mergedReads.bam").exists() }
            )
        }

    }

}
