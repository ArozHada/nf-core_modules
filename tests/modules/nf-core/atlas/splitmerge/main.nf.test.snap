{
    "Human Illumina PE Sorted BAM": {
        "content": [
            "versions.yml:md5,208f6ef7923cec00ddbcd5825f84a4fa",
            "test.paired_end.sorted_ignoredReads.txt.gz:md5,9b64c47313d2de89c26790f713707ee6"
        ],
        "timestamp": "2023-04-07T21:09:54+0000"
    }
}